awx/api/authentication.py

```diff
@@ -137,3 +32,28 @@ def authenticate_header(self, request):
         if not settings.AUTH_BASIC_ENABLED:
             return
         return super(LoggedBasicAuthentication, self).authenticate_header(request)
+
+
+class SessionAuthentication(authentication.SessionAuthentication):
+    
+    def authenticate_header(self, request):
+        return 'Session'
+
+    def enforce_csrf(self, request):
+        return None
+
+
+class LoggedOAuth2Authentication(OAuth2Authentication):
+
+    def authenticate(self, request):
+        ret = super(LoggedOAuth2Authentication, self).authenticate(request)
+        if ret:
+            user, token = ret
+            username = user.username if user else '<none>'
+            logger.debug(smart_text(
+                u"User {} performed a {} to {} through the API using OAuth token {}".format(
+                    username, request.method, request.path, user
```

Comment:
Did you pass in the wrong final argument to `.format`? You gave `user`, but this would be redundant with `username`, that should probably be the token. But what happens when you stringify a token? I hope we aren't writing the private tokens to the logs!

Reply:
I double checked and we are not exposing token values in the logs here.  Instead of user, it passes the token's <pk> now.